lib/events/auditwriter.go

```diff
@@ -139,6 +145,32 @@ func (cfg *AuditWriterConfig) CheckAndSetDefaults() error {
 	if cfg.BackoffDuration == 0 {
 		cfg.BackoffDuration = defaults.NetworkBackoffDuration
 	}
+	if cfg.MakeEvents == nil {
```

Comment:
Note: we technically don't need this, as it's only used when you use `AuditWriter`'s `io.Writer` interface, and we are calling `EmitAuditEvent()` directly.

However, it felt like a trap to have a general purpose tool called `AuditWriter` that had special case logic for SSH sessions and didn't provide a way to handle desktop or other types of sessions.

I could be convinced to remove this. Longer term, we should probably have a separate type that wraps `AuditWriter` and provides the `io.Writer` interface where necessary.

Reply:
I don't especially mind this but it might be nice to keep the default behavior in a named function rather than embedded in a closure here, needing to keep a field `nil` to select a particular implementation seems like an unusual API convention.

Reply:
Seems reasonable to me too, although I'm not intimately familiar with these parts. A named func would be nice (and testable!)